plugins/org.jkiss.dbeaver.ext.dameng/src/org/jkiss/dbeaver/ext/dameng/model/DamengMetaModel.java

```diff
@@ -227,9 +231,71 @@ public List<? extends GenericTrigger> loadTriggers(DBRProgressMonitor monitor, G
         }
     }
 
+    @Override
+    public void loadProcedures(DBRProgressMonitor monitor, GenericObjectContainer container) throws DBException {
+        GenericDataSource dataSource = container.getDataSource();
+        try (JDBCSession session = DBUtils.openMetaSession(monitor, container, "Read Dameng procedure source")) {
+            try (JDBCPreparedStatement dbStat = session.prepareStatement(
+                    "SELECT\n" +
+                            "PROC_OBJ.ID,\n" +
+                            "PROC_OBJ.NAME,\n" +
+                            "PROC_OBJ.CRTDATE,\n" +
+                            "PROC_OBJ.INFO1,\n" +
+                            "PROC_OBJ.VALID\n" +
+                            "FROM\n" +
+                            "SYSOBJECTS PROC_OBJ,\n" +
+                            "SYSOBJECTS SCH_OBJ,\n" +
+                            "SYSOBJECTS USER_OBJ\n" +
+                            "WHERE\n" +
+                            "PROC_OBJ.SCHID = SCH_OBJ.ID\n" +
+                            "AND SCH_OBJ.PID = USER_OBJ.ID\n" +
+                            "AND PROC_OBJ.SUBTYPE$ = 'PROC'\n" +
```

Comment:
`PROC_OBJ.SUBTYPE$ = 'PROC'`

Does it mean that functions will be also in the query result?

Reply:
Yes, contains function , distinguished by info1

Reply:
Ok